docs/source/recipes/plot_14_recipe.py

```diff
@@ -0,0 +1,181 @@
+"""
+Overlay Geopotential height contours over Temperature anomalies
+===============================================================
+
+In this recipe, we will overlay Geopotential height contours over Temperature
+anomalies to help analyse meteorological conditions during July 2018,
+specifically focusing on the significant concurrent extreme events that occurred
+during the 2018 boreal spring/summer season in the Northern Hemisphere.
+
+"""
+
+# %%
+# 1. Import cf-python and cf-plot:
+import cfplot as cfp
+
+import cf
+
+# %%
+# 2. Read and select the 200 hpa geopotential by index and look at its contents:
+gp = cf.read("~/recipes/ERA5_monthly_averaged_z200.nc")[0]
+print(gp)
+
+# %%
+# 3. Convert the geopotential data to geopotential height by dividing it by the
+# acceleration due to gravity (approximated as 9.81 m/s^2):
+gph = gp / 9.81
+
+# %%
+# 4. Subset the geopotential height to extract data specifically for July 2018,
+# a significant month due to heat extremes and heavy rainfall:
+gph_july = gph.subspace(T=cf.month(7) & cf.year(2018)).squeeze()
+
+# %%
+# 5. Plot contour lines of this geopotential height for July 2018. Here:
+#
+# - `cfplot.gopen <http://ajheaps.github.io/cf-plot/gopen.html>`_ is used to
+#   define the parts of the plot area, which is closed by
+#   `cfplot.gclose <http://ajheaps.github.io/cf-plot/gclose.html>`_;
+# - `cfplot.mapset <https://ajheaps.github.io/cf-plot/mapset.html>`_ is used to
+#   set the map projection to North Polar Stereographic;
+# - `cfplot.setvars <http://ajheaps.github.io/cf-plot/setvars.html>`_ is used to
+#   set various attributes of the plot, like setting the thickness of the lines
+#   that represent continents;
+# - `cfplot.con <http://ajheaps.github.io/cf-plot/con.html>`_ plots the contour
+#   lines representing the 200 hpa geopotential height values without filling
+#   between the contour lines (``fill=False``) and no colour bar
+#   (``colorbar=False``);
+# - `cfplot.levs <https://ajheaps.github.io/cf-plot/levs.html>`_ is used to
+#   specify two contour levels, 12000 and 12300 m, corresponding to the
+#   approximate polar-front jet and subtropical jet respectively;
+# - `cfplot.con <http://ajheaps.github.io/cf-plot/con.html>`_ is again used to
+#   plot the contour lines for polar-front jet and subtropical jet with a
+#   thicker line width;
+# - `cfp.plotvars.mymap.stock_img() <https://scitools.org.uk/cartopy/docs/v0.15/matplotlib/geoaxes.html#cartopy.mpl.geoaxes.GeoAxes.stock_img>`_
+#   then finally visualises the Earth's surface in cf-plot's
+#   ``cfp.plotvars.mymap`` plot object:
```

Comment:
This seems more likely to be a cf-plot issue, but I noticed something slightly strange with the plot that emerges here, namely that the outline is off from the background Earth image, most noticeable in the top left corner. Do you know why that is happening (if not I'll check in with Andy)? It would be nice to get it rectified since it makes me question the validity of the plotting if the outline seems out of place...
![line_out_cfplot](https://github.com/NCAS-CMS/cf-python/assets/30274190/a7381eba-7761-4815-adf9-9c1f543e10df)

Reply:
I had thought it has to do with the `cfp.plotvars.mymap.stock_img()`, but then when I tried using `cfp.plotvars.mymap.add_feature(cfeature.OCEAN)` or `cfp.plotvars.mymap.add_feature(cfeature.LAND)`, it still gives the similar outline issue:

```python
import cartopy.feature as cfeature

cfp.gopen()
cfp.mapset(proj="npstere")
cfp.setvars(continent_thickness=0.5)

cfp.con(
    f=gph_july,
    fill=False,
    lines=True,
    line_labels=False,
    colors="black",
    linewidths=1,
    colorbar=False,
)

cfp.levs(manual=[12000, 12300])
cfp.con(
    f=gph_july,
    fill=False,
    lines=True,
    colors="black",
    linewidths=3.0,
    colorbar=False,
)


cfp.plotvars.mymap.add_feature(cfeature.OCEAN)
cfp.plotvars.mymap.add_feature(cfeature.LAND)

cfp.gclose()
```
![image](https://github.com/NCAS-CMS/cf-python/assets/83757805/3333858f-9220-4646-ae7e-550b24749ec2)

Reply:
Thanks for sharing your findings here. Hmm, I'll report this to Andy, if you haven't already, since this could be a bug.

Reply:
As for this PR, it's not related to your code, so I'll mark this as resolved.